src/keytar_win.cc

```diff
@@ -19,6 +20,8 @@ KEYTAR_OP_RESULT SetPassword(const std::string& service,
                  const std::string& password,
                  std::string* errStr) {
   std::string target_name = service + '/' + account;
+  static std::mutex mutex;
+  std::lock_guard<std::mutex> lock(mutex);
```

Comment:
since this could be perf intensive, please also add a comment.
consider making this mutex lock a switch (additional argument, or new method overload) that consumers can use to bypass if they dont use the API in multi threaded environments.